File: slack-reader/src/main/java/dev/number6/slackreader/dagger/SlackReaderComponent.java
package dev.number6.slackreader.dagger;

import dev.number6.db.dagger.DatabaseConfigurationPortModule;
import dev.number6.db.dagger.DatabasePortModule;
import dev.number6.db.dagger.DynamoDBMapperModule;
import dev.number6.slack.dagger.AWSSecretsModule;
import dev.number6.slack.dagger.HttpModule;
import dev.number6.slackreader.SlackReader;
import dagger.Component;

@Component(modules = {SlackServiceModule.class,
        SlackPortModule.class,
        HttpModule.class,
        DatabasePortModule.class,
        DatabaseConfigurationPortModule.class,
        DynamoDBMapperModule.class,
        AmazonSnsModule.class,
        ConfigurationPortModule.class,
        ClockModule.class})
public interface SlackReaderComponent {

    SlackReader handler();
}

File: slack-reader/src/main/kotlin/dev/number6/slackreader/dagger/SlackReaderComponent.kt
package dev.number6.slackreader.dagger

import dagger.Component
import dev.number6.db.dagger.DatabaseConfigurationPortModule
import dev.number6.db.dagger.BasicDatabasePortModule
import dev.number6.db.dagger.DynamoDBMapperModule
import dev.number6.slack.dagger.SlackClientLiveBindingsModule
import dev.number6.slackreader.SlackReader

@Component(modules = [SlackServiceModule::class,
    SlackReaderPortModule::class,
    SlackClientLiveBindingsModule::class,
    BasicDatabasePortModule::class,
    DatabaseConfigurationPortModule::class,
    DynamoDBMapperModule::class,
    AmazonSnsModule::class,
    ConfigurationPortModule::class,
    ClockModule::class])
interface SlackReaderComponent {
    fun handler(): SlackReader
}